src/Bicep.LangServer/Deploy/DeploymentHelper.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Bicep.Core;
+using Bicep.Core.Json;
+
+namespace Bicep.LanguageServer.Deploy
+{
+    public class DeploymentHelper
+    {
+        /// <summary>
+        /// Creates a deployment at provided target scope and returns deployment succeeded/failed message.
+        /// </summary>
+        /// <param name="deploymentCollectionProvider">deployment collection provider</param>
+        /// <param name="armClient">arm client</param>
+        /// <param name="template">template used in deployment</param>
+        /// <param name="parameterFilePath">path to parameter file used in deployment</param>
+        /// <param name="id">id string to create the ResourceIdentifier from</param>
+        /// <param name="scope">target scope</param>
+        /// <param name="location">location to store the deployment data</param>
+        /// <returns>deployment succeeded/failed message</returns>
+        public static async Task<string> CreateDeployment(
+            IDeploymentCollectionProvider deploymentCollectionProvider,
+            ArmClient armClient,
+            string template,
+            string parameterFilePath,
+            string id,
+            string scope,
+            string location)
+        {
+            if ((scope == LanguageConstants.TargetScopeTypeSubscription ||
+                scope == LanguageConstants.TargetScopeTypeManagementGroup) &&
+                string.IsNullOrWhiteSpace(location))
+            {
+                return LangServerResources.MissingLocationDeploymentFailedMessage;
+            }
+
+            DeploymentCollection? deploymentCollection;
+            var resourceIdentifier = new ResourceIdentifier(id);
+
+            try
+            {
+                deploymentCollection = deploymentCollectionProvider.GetDeploymentCollection(armClient, resourceIdentifier, scope);
```

Comment:
Should GetDeploymentCollection be able to return null?

Reply:
Hmmm maybe not. Updated here:
https://github.com/Azure/bicep/commit/67ff4dea5d71e3f1f1736f79cc89290e85857133

Thanks!